python/ray/_raylet.pyx

```diff
@@ -699,7 +747,7 @@ cdef class CoreWorker:
             raylet_socket.encode("ascii"), job_id.native(),
             gcs_options.native()[0], log_dir.encode("utf-8"),
             node_ip_address.encode("utf-8"), task_execution_handler,
-            check_signals, False))
+            check_signals, True))
```

Comment:
Were the some concerns enabling the memory store by default?

Reply:
Oh yes this will cause some problems. The issue is that object IDs weren't being properly generated so they would randomly be put/got from the wrong object store.

Reply:
Do you know where they are being generated? We can just retry if they happen to be a direct actor call id.

Reply:
Addressed by #6003